crates/miden-lib/asm/kernels/transaction/api.masm

```diff
@@ -836,6 +840,139 @@ export.input_note_get_metadata
     # => [METADATA, pad(12)]
 end
 
+#! Returns the serial number of the specified input note.
+#!
+#! Inputs:  [is_current_note, note_index, pad(14)]
+#! Outputs: [SERIAL_NUMBER, pad(12)]
+#!
+#! Where:
+#! - is_current_note is the boolean flag indicating whether we should return the serial number
+#!   of the currently processing note or of the note with the specified index.
+#! - note_index is the index of the input note whose serial number should be returned. Notice that 
+#!   if is_current_note flag is true (we should process the currently processed note), this index is
+#!   assumed to be 0 (pad element).
+#! - SERIAL_NUMBER is the serial number of the specified input note.
+#!
+#! Panics:
+#! - if requested by note index: the note index is greater or equal to the total number of input 
+#!   notes.
+#! - if requested from current note: the input note is not being processed (attempted to access note
+#!   inputs from incorrect context).
+#!
+#! Invocation: dynexec
+export.input_note_get_serial_number
+    # get the input note pointer depending on whether the requested note is current or it was 
+    # requested by index.
+    exec.get_requested_note_ptr
+    # => [input_note_ptr, pad(15)]
+
+    # assert the pointer is not zero - this would suggest the procedure has been called from an
+    # incorrect context
+    dup neq.0 assert.err=ERR_NOTE_ATTEMPT_TO_ACCESS_NOTE_SERIAL_NUMBER_FROM_INCORRECT_CONTEXT
+    # => [input_note_ptr, pad(15)]
+
+    # get the serial number using the note pointer
+    exec.memory::get_input_note_serial_num
+    # => [SERIAL_NUMBER, pad(15)]
+
+    # truncate the stack
+    repeat.3
+        movup.4 drop
+    end
+    # => [SERIAL_NUMBER, pad(12)]
+end
+
+#! Returns the inputs commitment and length of the specified input note.
+#!
+#! Inputs:  [is_current_note, note_index, pad(14)]
+#! Outputs: [NOTE_INPUTS_COMMITMENT, num_inputs, pad(11)]
+#!
+#! Where:
+#! - is_current_note is the boolean flag indicating whether we should return the inputs commitment 
+#!   and length from the currently processing note or from the note with the specified index.
+#! - note_index is the index of the input note whose data should be returned. Notice that if
+#!   is_current_note flag is true (we should process the currently processed note), this index is
+#!   assumed to be 0 (pad element).
+#! - NOTE_INPUTS_COMMITMENT is the inputs commitment of the specified input note.
+#! - num_inputs is the number of input values of the specified input note.
+#!
+#! Panics:
+#! - if requested by note index: the note index is greater or equal to the total number of input 
+#!   notes.
+#! - if requested from current note: the input note is not being processed (attempted to access note
+#!   inputs from incorrect context).
+#!
+#! Invocation: dynexec
+export.input_note_get_data_info
```

Comment:
With https://github.com/0xMiden/miden-base/issues/1662 we'd have to name this `input_note_get_payload_info`, and I guess we can already do this.